sudoku.py

```
@@ -174,6 +174,14 @@ def main():
     for label in solution_list:
         coord, digit = label.split('_')
         row, col = map(int, coord.split(','))
+
+        if matrix[row][col] > 0:
+            # the returned solution is not optimal and either tried to
+            # overwrite one of the starting values, or returned more than
+            # one value for the position. In either case the solution is
+            # likely incorrect.
```

Maybe raise a warning as well?

---

Is it worth checking the result differs from the initial (fixed) value?

---

Could do either. If I was writing it from scratch I would raise an exception in the failing case rather than just printing "the solution is incorrect". But IMO this was the least invasive change relative to the original design.

---

Ok, we can keep it simple then.